engine/src/dskmac.cpp

```diff
@@ -4627,9 +4628,75 @@ struct MCMacDesktop: public MCSystemInterface, public MCMacSystemService
         if (t_cf_document != NULL)
             CFRelease(t_cf_document);
     }
-    
+
+#define APPLESCRIPT_SCRIPT \
+	"local tTempFolder;" \
+	"put the tempname into tTempFolder;" \
+	"create folder tTempFolder;" \
+	"local tStdout, tStderr, tScript;" \
+	"put tTempFolder & \"/stdout.txt\" into tStdout;" \
+	"put tTempFolder & \"/stderr.txt\" into tStderr;" \
+	"put tTempFolder & \"/script.scpt\" into tScript;" \
+	"put textEncode(param(1), \"utf8\") into url (\"binfile:\" & tScript);" \
+	"get shell(format(\"arch -arm64 osascript %s 2>%s 1>%s\", tScript, tStderr, tStdout));" \
+	"local tResult;" \
+	"put the result into tResult;" \
+	"delete file tScript;" \
+	"local tValue;" \
+	"if tResult is 0 then;" \
+		"put url (\"binfile:\" & tStdout) into tValue;" \
+		"if the last char of tValue is return then;" \
+			"delete the last char of tValue;" \
+		"end if;" \
+	"else;" \
+		"put \"execution error\" into tValue;" \
+	"end if;" \
+	"delete file tStdout;" \
+	"delete file tStderr;" \
+	"switch tValue;" \
+	"case \"execution error\";" \
+	"case empty;" \
+		"return tValue;" \
+	"default;" \
+		"return \"{\" & tValue & \"}\";" \
+	"end switch"
+
+	int process_is_translanslated()
```

Comment:
I suggest
```
static bool GetProcessIsTranslated(void)
{
    static int s_state = -1;
    if (s_state == -1)
    {
        ....
    }
    return s_state == 1;
}
```

The ordering of the stuff in this file is a little bit messy, but the definition shouldn't be in the class, maybe put it at the top where you currently define the file-local static.

(Rationale for putting the static in the function is that it encapsulates it better).